### PR TITLE
1622 table of rois

### LINKED ISSUE #1622: A table exists to display ROIs created taking design inspiration from the reconstruction window COR table

### PR DESCRIPTION
**Pull request by JackEAllen**
### Issue

Closes #1622 

### Description

This PR Adds the following functionality to the ROI table within spectrum viewer window to improve user workflow using multiple ROIs:
* Populate table with visible ROIs
* Allow user to add new ROIs
* Allow user to rename ROIs (double click on name)
* Allow user to clear selected ROI
* Default ROIs cannot be modified
* ROIs cannot have duplicate names or be empty e.g. ""
* The most recently added ROIs ToF data is plotted (future PR will allow for plotting of multiple ROI ToF simultaneously)

### Testing 
- [ ] Run Unit tests and ensure they pass
- [ ] Ensure code is cleanly implemented and readable
- [ ] All action buttons within table perform as expected

### Acceptance Criteria 
- [ ] Unit test pass (`mantidimaging/gui/windows/spectrum_viewer/test/model_test.py` and `mantidimaging/gui/windows/spectrum_viewer/test/presenter_test.py`)
- [ ] New ROIs can be added and are populated in table using "Add" button
- [ ] ROIs can be renamed
- [ ] Selected ROI can excluding default ROI can be cleared using "Remove"
- [ ] Renamed ROIs are updated throughout application and displayed when exporting data to .csv

### Documentation

*How have you changed the documentation to reflect your changes? All changes should be noted in the appropriate file in docs/release_notes*

### CODE REVIEW

File: mantidimaging/gui/windows/spectrum_viewer/test/model_test.py [hunk 1]
```diff
@@ -232,3 +232,42 @@ def test_WHEN_get_roi_called_with_non_existent_name_THEN_error_raised(self):
...
+        self.model.set_new_roi("new_roi")
+        self.assertEqual(self.model.get_list_of_roi_names(), ["all", "roi", "new_roi"])
+        self.model.remove_roi("new_roi")
+        self.assertEqual(self.model.get_list_of_roi_names(), ["all", "roi"])
+
+    def test_WHEN_remove_roi_called_with_default_roi_THEN_raise_runtime_error(self):
+        self.model.set_stack(generate_images())
+        with self.assertRaises(RuntimeError):
+            self.model.remove_roi("all")
+            self.assertEqual(self.model.get_list_of_roi_names(), ["all", "roi"])
```
[the remark this answers is not in the document]
Comment (JackEAllen) [reply mantidimaging/gui/windows/spectrum_viewer/test/model_test.py]: Thanks for catching this one, Should be outside of keyerror assertion now

**Review by samtygier-stfc — CHANGES_REQUESTED** [review]
Can I also suggest removing the "Clear All" button. I don't think we really need it, and may end up short for space later if we add more features.

**Comment by JackEAllen** [comment]
> Can I also suggest removing the "Clear All" button. I don't think we really need it, and may end up short for space later if we add more features.

Would you like me to remove just the button or button and relevant methods if this is not needed at all?

**Comment by samtygier-stfc** [comment]
> Would you like me to remove just the button or button and relevant methods if this is not needed at all?

Lets remove the methods, we can always recover them from git if the feature is requested. (Unless you think they will been needed when handling a stack change).

**Comment by JackEAllen** [comment]
> > Would you like me to remove just the button or button and relevant methods if this is not needed at all?
> 
> Lets remove the methods, we can always recover them from git if the feature is requested. (Unless you think they will been needed when handling a stack change).

I imagine they should be fine to remove if there's no use case for users to remove all ROIs at once

**Review by samtygier-stfc — APPROVED** [review]

**Comment by samtygier-stfc** [comment]
Only failing tests are due to new screenshots on another PR that has not landed yet.

**Pull request merged**